timemachine/cpp/src/centroid_restraint.cu

```diff
@@ -20,7 +20,7 @@ CentroidRestraint<RealType>::CentroidRestraint(
     cudaSafeMalloc(&d_centroid_a_, 3 * sizeof(*d_centroid_a_));
     cudaSafeMalloc(&d_centroid_b_, 3 * sizeof(*d_centroid_b_));
 
-    cudaSafeMalloc(&d_u_buffer_, (N_A_ + N_B_) * sizeof(d_u_buffer_));
```

Comment:
Previously I had read the kernels incorrectly, so this was `N_A + N_B`, but really it is only a single energy value. 

Also notice `sizeof(d_u_buffer)` rather than `sizeof(*d_u_buffer)`, which previously worked because the size of the pointer and the value was identical. Not the case with int128